libs/arcade-evals/arcade_evals/loaders.py

```diff
@@ -0,0 +1,361 @@
+            request_headers["Accept"] = "application/json, text/event-stream"
+
+        try:
+            response = httpx.post(
+                url,
+                json={"jsonrpc": "2.0", "id": 1, "method": "tools/list", "params": {}},
+                headers=request_headers,
+                timeout=timeout,
+            )
+            response.raise_for_status()
+            data = response.json()
+            if "result" in data and "tools" in data["result"]:
+                result_tools: list[dict[str, Any]] = data["result"]["tools"]
+                return result_tools
+        except httpx.HTTPStatusError as e:
+            # Fallback to SSE if server requires it
+            if e.response.status_code == 406 and "text/event-stream" in e.response.text:
+                return load_from_http(url=url, timeout=timeout, headers=headers, use_sse=True)
+            # Some MCP servers (e.g., gateways) require an initialize/session handshake
+            # before handling tools/list. If we received an auth/Bad Request style error
+            # and do not yet have a session header, try the session handshake flow.
+            if (
+                e.response.status_code in (400, 401, 403)
+                and "Mcp-Session-Id" not in request_headers
+            ):
+                try:
+                    return _load_with_session(url=url, headers=request_headers, timeout=timeout)
+                except Exception:
+                    return []
+        except (
+            httpx.TimeoutException,
+            httpx.ConnectError,
+            json.JSONDecodeError,
+            httpx.UnsupportedProtocol,
+        ):
+            # Timeout/unreachable/invalid response - return empty list (graceful degradation)
+            return []
+
+    return []
+
+
+def load_arcade_cloud(
+    gateway_slug: str,
+    arcade_api_key: str | None = None,
+    arcade_user_id: str | None = None,
+    timeout: int = 10,
+) -> list[dict[str, Any]]:
+    """
+    Load tools from Arcade Cloud MCP gateway via HTTP.
+
+    Connects to https://api.arcade.dev/mcp/<gateway_slug>
+
+    Args:
+        gateway_slug: Your gateway slug (e.g., "my-gateway").
+        arcade_api_key: API key (or set ARCADE_API_KEY env var).
+        arcade_user_id: User ID (or set ARCADE_USER_ID env var).
+        timeout: Request timeout in seconds.
+
+    Returns:
+        List of tool descriptors. Empty list on error.
+
+    Example:
+        >>> tools = load_arcade_cloud(
+        ...     gateway_slug="my-gateway",
+        ...     arcade_api_key="arc_...",
+        ...     arcade_user_id="user@example.com"
+        ... )
+    """
+    api_key = arcade_api_key or os.environ.get("ARCADE_API_KEY")
+    user_id = arcade_user_id or os.environ.get("ARCADE_USER_ID")
+
+    headers: dict[str, str] = {"Accept": "application/json, text/event-stream"}
+    if api_key:
+        headers["Authorization"] = api_key
+    if user_id:
+        headers["arcade-user-id"] = user_id
```

Comment:
I believe connecting to Arcade Gateways with these headers is going to be deprecated in the very near future. And the new way will require arcade_evals (the MCP client in this case) to be able to talk to MCP severs that are protected by an Auth Server.